app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,75 @@
+class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: [:public]
+  before_action :set_recipe, only: %i[show edit update destroy]
+
+  def public
+    @recipes = Recipe.where(public: true)
+  end
+
+  # GET /recipes or /recipes.json
+  def index
+    @recipes = Recipe.where(user_id: current_user.id)
+  end
+
+  # GET /recipes/1 or /recipes/1.json
+  def show; end
```

Comment:
Please add a toggle button on the right side of the recipe show page which can be seen in the given [wireframe](https://raw.githubusercontent.com/microverseinc/curriculum-rails/main/recipe-app/images/recipe_food.png?token=GHSAT0AAAAAABT3TAGW4NEBNFZ3FJJVVKXGYUGTZIQ).